inlong-agent/agent-core/src/main/java/org/apache/inlong/agent/core/HeartbeatManager.java

```diff
@@ -70,13 +88,65 @@ public HeartbeatManager(AgentManager agentManager) {
         jobmanager = agentManager.getJobManager();
         httpManager = new HttpManager(conf);
         baseManagerUrl = buildBaseUrl();
-        reportSnapshotUrl = builReportSnapShotUrl(baseManagerUrl);
+        reportSnapshotUrl = buildReportSnapShotUrl(baseManagerUrl);
+        reportHeartbeatUrl = buildReportHeartbeatUrl();
+    }
+
+    @Override
+    public void start() throws Exception {
+        submitWorker(snapshotReportThread());
+        submitWorker(heartbeatReportThread());
+    }
+
+    private Runnable snapshotReportThread() {
+        return () -> {
+            while (isRunnable()) {
+                try {
+                    TaskSnapshotRequest taskSnapshotRequest = buildTaskSnapshotRequest();
+                    httpManager.doSentPost(reportSnapshotUrl, taskSnapshotRequest);
+                    if (LOGGER.isDebugEnabled()) {
+                        LOGGER.debug(" {} report to manager", taskSnapshotRequest);
+                    }
+                    int heartbeatInterval = conf.getInt(AGENT_HEARTBEAT_INTERVAL,
```

Comment:
the variable 'heartbeatInterval' is only used once. I don't think it is necessary to define the variable.

Reply:
I think it is